File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/ILoopMax.cs
using System;
using System.Collections.Generic;

namespace DFIR_Compiler
{
    public class ILoopMax:IBorderNode
    {

        public ILoopMax(int nodeId, int parentId, List<ITerminal> inputTerminals, List<ITerminal> outputTerminals) : base(nodeId, parentId, inputTerminals, outputTerminals)
        {
            INodeType = "ILoopMax";
        }


        public override void DecreaseIndegree()
        {
            indegree--;
            IdToINode[parentId].indegree--;
        }

        public override void Print()
        {
            Console.WriteLine("------------------------------------");
            Console.WriteLine("ILoopMax:");
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
            for (int i = 0; i < inputTerminals.Count; i++)
            {
                inputTerminals[i].Print();
            }
            for (int i = 0; i < outputTerminals.Count; i++)
            {
                outputTerminals[i].Print();
            }
        }
    }
}

File: Wayne/WayneCompiler/src/Dfir_representation/ILoopMax.java
package Dfir_representation;

import java.util.Vector;


public class ILoopMax extends IBorderNode{
	   
	   public ILoopMax(int nodeId, int parentId, Vector<ITerminal> inputTerminals,Vector<ITerminal> outputTerminals){
		   super(nodeId,parentId,inputTerminals,outputTerminals);
		   JNodeType = "ILoopMax";
	   }
	   public void decIndegree(){
		   this.indegree--;
		   JNode.IdToJNode.get(this.parentId).indegree--;
	   }
	   
	   public void print(){
		   System.out.println("ILoopMax begin");
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
		   for(int i=0; i<inputTerminals.size(); i++){
				   inputTerminals.get(i).print();
		   }
		  for(int i=0; i<outputTerminals.size(); i++){
				   outputTerminals.get(i).print();
		   }   
		  System.out.println("ILoopMax end\n");
	   }
}